tools/check-imports.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import glob
+import os
+import re
+import sys
+
+
+def do_exist(file, lines, imported):
+  if not any(not re.match('using \w+::{0};'.format(imported), line) and
+             re.search(imported, line) for line in lines):
+    print('File "{0}" does not use "{1}"'.format(file, imported))
+    return False
+  return True
+
+
+def is_valid(file):
+  with open(file) as source_file:
+    lines = [line.strip() for line in source_file]
+
+  usings, importeds, line_numbers, valid = [], [], [], True
+  for idx, line in enumerate(lines, 1):
+    matches = re.search(r'^using (\w+::(\w+));$', line)
+    if matches:
+      line_numbers.append(idx)
+      usings.append(matches.group(1))
+      importeds.append(matches.group(2))
+
+  valid = all([do_exist(file, lines, imported) for imported in importeds])
+
+  sorted_usings = sorted(usings, key=lambda x: x.lower())
+  if sorted_usings != usings:
+    print("using statements aren't sorted in '{0}'.".format(file))
+    for num, actual, expected in zip(line_numbers, usings, sorted_usings):
+      if actual != expected:
+        print('\tLine {0}: Actual: {1}, Expected: {2}'
+            .format(num, actual, expected))
+    return False
+  else:
+    return valid
+
+sys.exit(0 if all([is_valid(file) for file in glob.iglob('src/*.cc')]) else 1)
+
```

Comment:
@bnoordhuis Actually, that is necessary. `all` and `any` short-circuit. They return `True` and `False` the moment they meet an item doesn't meet the criteria. Passing a generator expression will leave the rest of them unexecuted. For example,

```py
>>> gen = (num for num in xrange(10))
>>> all(item * item < 20 for item in gen)
False
>>> list(gen)
[6, 7, 8, 9]
```

But when we use List Comprehension, it executes all of them and only their results are used with `all` and `any`. For example,

```py
>>> gen = (num for num in xrange(10))
>>> all([item * item < 20 for item in gen])
False
>>> list(gen)
[]
```

Wrote more about it in [this StackOverflow answer](http://stackoverflow.com/a/19389957/1903116)